week-2/extended-project/yoda-y-sus-apis/web_api_methods.rb

```diff
@@ -0,0 +1,37 @@
+require "http"
+require "json"
+require "csv"
+
+$index_general = 0
+$uri = "https://www.cnet.com/es/noticias/ranking-peliculas-star-wars-guerra-de-las-galaxias/"
+$response = ""
+def initialization_scrapping
+  $response = HTTP.get($uri).body.to_s
+  text_references = "Ránking de las películas de 'Star Wars' según CNET en Español"
+  $index_general = /#{text_references}/ =~ $response
+end
+
+def get_score(name_movie)
+  puts name_movie
+  index_movie = $response.index(name_movie, $index_general)
+  index_rating = $response.index(">:", index_movie) + 3
+  $response[index_rating..(index_rating + 2)].gsub("<","").to_i.to_s
+end
+
+def get_data(page)
+    data_next = []
+    request = HTTP.get(page)
+    data = JSON.parse(request.to_s)["results"]
+    page_number = JSON.parse(request.to_s)["next"]
+    if page_number
+      data_next = get_data(page_number)
```

Comment:
Why do you use a recursive method?

Reply:
Because the data of the api is divided in page, when you get the data, it show a key named "Next" to indicate us that the rest of data is in another pages and we need use all data provided for the api.